proposals/devcontainer-features.md

```diff
@@ -0,0 +1,143 @@
+# Dev container features reference
+
+Dev container features provide a smooth path for customizing your container definitions. Features are self contained units of code meant to facilitate creation of custom images for application or development containers.
+
+From a practical point of view, features are folders that contain units of code with different entrypoints for different lifecycle events.
+
+Features can be defined by a `feature.json` file in the root folder of the feature. The file is optional for backwards compatibility but it is required for any new features being authored.
+
+Features are to be executed in sequence as defined in `devcontainer.json`.
+
+## Folder Structure 
+
+A feature is a self contained entity in a folder. A feature release would be a tar file that contains all files part of the feature.
+
++-- feature
+|    +-- feature.json
+|    +-- validate.sh (default)
+|    +-- aquire.sh (default)
+|    +-- install.sh (default)
+|    +-- (other files)
+
+In case `feature.json`  does not include a reference for the lifecycle scripts the application will look for the default script names and will execute them if available.
+
+In case there is intent to create a set of features that share code, it is possible to create a feature collection in the following way:
+
+collectionFolder
++-- feature-collection.json
```

Comment:
I liked the idea of being able to distribute definitions/templates and features together. Should we make this a `devcontainer-definitions.json` to contain templates and features?

Reply:
I have `devcontainer-collection.json` here: https://github.com/microsoft/vscode-dev-containers/blob/main/devcontainer-collection.json

Reply:
Changed to `devcontainer-collection` to signify that its a group.

Reply:
Does that make it sound like it is a collection of dev containers. Would `devcontainer-definitions.json` go as the abbreviation for `devcontainer-templates-and-features.json`?

Reply:
I like the idea of `devcontainer-collection.json` since that way the file can contain both templates and features related to `devcontainers` as users know them.